src/var.c

```diff
@@ -53,8 +55,37 @@ struct fi_var_t {
 
 static struct fi_var_t *fi_vars = NULL;
 
+__attribute__((visibility ("default")))
+int DEFAULT_SYMVER_PRE(fi_getvars)(struct fi_env **vars, int *count) {
+
+	struct fi_env *vhead;
+	int len = 0;
+
+	// just get a count
+	for (struct fi_var_t *ptr = fi_vars; ptr ;ptr = ptr->next, ++len) {}
+
+	vhead = calloc(len, sizeof (*vhead));
+
+	if (!vhead)
+		return -FI_ENOMEM;
+
+	int i = 0;
+	for (struct fi_var_t *ptr = fi_vars; ptr; ptr = ptr->next, ++i) {
+		vhead[i] = (struct fi_env) {	.provider = ptr->provider,
+						.var_name = ptr->var_name,
+						.env_var_name = ptr->env_var_name,
+						.help_string = ptr->help_string
+					   };
```

Comment:
Is there a reason why the code is structure like this, versus just doing:

vars[i].var_name = ptr->var_name;
vars[i].env_var_name = ptr->env_var_name;
...

Note that this is still a racy API.  This returns pointers to internal structures, which means that the library can never free this memory after this call is made.  Use strdup() instead.  And add an fi_freevars() call